core/src/main/java/jenkins/security/ResourceDomainRootAction.java

```diff
@@ -0,0 +1,307 @@
+        }
+
+        public void doDynamic(StaplerRequest req, StaplerResponse rsp) throws IOException {
+            String restOfPath = req.getRestOfPath();
+
+            String requestUrlSuffix = this.browserUrl;
+
+            LOGGER.fine(() -> "Performing a request as authentication: " + authenticationName + " and restOfUrl: " + requestUrlSuffix + " and restOfPath: " + restOfPath);
+
+            Authentication auth = Jenkins.ANONYMOUS;
+            if (authenticationName != null) {
+                User user = User.getById(authenticationName, false);
+                if (user != null) {
+                    try {
+                        auth = user.impersonate();
+                        LOGGER.fine(() -> "Successfully impersonated " + authenticationName);
+                    } catch (UsernameNotFoundException ex) {
+                        LOGGER.log(Level.FINE, "Failed to impersonate " + authenticationName, ex);
+                        rsp.sendError(403, "No such user: " + authenticationName);
+                        return;
+                    }
+                }
+            }
+
+            try (ACLContext ignored = ACL.as(auth)) {
+                try {
+                    Stapler.getCurrent().invoke(req, rsp, Jenkins.get(), requestUrlSuffix + restOfPath);
+                } catch (Exception ex) {
+                    // cf. UnwrapSecurityExceptionFilter
+                    Throwable cause = ex.getCause();
+                    while (cause != null) {
+                        if (cause instanceof AccessDeniedException) {
+                            throw (AccessDeniedException) cause;
+                        }
+                        cause = cause.getCause();
+                    }
+                    throw ex;
+                }
+                /*
+                While we could just redirect below to the real URL like we do for expired resource URLs, the question
+                is whether we'd end up in a redirect loop if the exception is specific to this mode (and the "normal"
+                URLs redirect to resource URLs). That seems even worse than an error here.
+                 */
+            } catch (AccessDeniedException ade) {
+                /* This is expected to be fairly common, as permission issues are thrown up as exceptions */
+                LOGGER.log(Level.FINE, "Failed permission check for resource URL access", ade);
+                rsp.sendError(403, "Failed permission check: " + ade.getMessage());
+            } catch (Exception e) {
+                /*
+                This should be fairly uncommon -- it's basically the 'rage' butler response. Notably, lack of access
+                to a job (permissions/deleted/renamed/...) would not throw an exception, but just sends a 404 response.
+                 */
+                LOGGER.log(Level.FINE, "Something else failed for resource URL access", e);
+                rsp.sendError(404);
```

Comment:
A generic 404 error is not terribly helpful when trying to debug problems. Especially when the logging associated w/ the error is likely to be be filtered out (FINE). I'd encourage the use of a distinct message.

Reply:
Trivial to increase the log level as needed.

As all of this is available without even Overall/Read, strictly speaking, published error content is deliberately minimal.